### PR TITLE
Improve generated client with more accurate response types

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR improves the slack client by adding various sanitations (via another Ballerina script) to the existing schema names in the OpenAPI contract. 

These changes will make sure that the Ballerina connector APIs consist of more accurate representation of the actual server responses in most of the cases.

Resolves https://github.com/ballerina-platform/ballerina-library/issues/6723

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: docs/spec/build.gradle [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+def files = ["sanitations.bal"];
```
Comment (ayeshLK) [inline docs/spec/build.gradle:21]: AFAIU the `sanitations.bal` should only be executed whenever we update the openapi-spec. So, I don't think we need to include this into the gradle build. I think we can add a gradle task which can be used to execute openapi sanitation on demand. @NipunaRanasinghe WDYT ?

File: docs/spec/sanitations.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ These changes are done in order to improve the overall usability, and to address
     * users.deletePhoto
     * users.setPhoto
 
-2. Changed the schema name `Default success template` to `DefaultSuccessResponse` in the components section of the OpenAPI specification, to improve the usability of the counterpart Ballerina record name.
+2. Sanitized the inline and component schema names by removing the whitespaces, special characters and, converting them to pascal case.
+   * This was done using the `sanitations.bal` script under the `docs/spec` directory.
```
Comment (ayeshLK) [inline docs/spec/sanitations.md:32]: Following up with this comment [1]; Here I think we can mention how we could execute the `sanitations.bal` using the gradle build.

[1] - https://github.com/ballerina-platform/module-ballerinax-slack/pull/164/files#r1674504405

File: docs/spec/build.gradle [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (AzeemMuzammil) [inline docs/spec/build.gradle:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
```

Shall we check the other places as well?

File: docs/spec/sanitations.bal [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                schema.title = sanitizedTitle;
+            }
+        }
+    }
+
+    return spec;
+}
+
+function sanitizeSchemaNames(Specification spec) returns string {
+
```
Comment (AzeemMuzammil) [inline docs/spec/sanitations.bal:118]:
```suggestion
```
do we need this extra line?

File: docs/spec/build.gradle (hunk 1 above)
Comment (NipunaRanasinghe) [reply docs/spec/build.gradle:21]: @ayeshLK please note the gradle build is added *only to build the sanitation script* during the project build, to make sure that the ballerina source code won't break with the upcoming language versions. And like you mentioned, running the script should be a one time thing, so it shouldn't be a part of the gradle build.

File: docs/spec/sanitations.bal (hunk 4 above)
Comment (NipunaRanasinghe) [reply docs/spec/sanitations.bal:118]: can remove it.

File: docs/spec/sanitations.md (hunk 2 above)
Comment (NipunaRanasinghe) [reply docs/spec/sanitations.md:32]: as per https://github.com/ballerina-platform/module-ballerinax-slack/pull/164#discussion_r1675072862, no need to mention it explicitly I guess.

File: docs/spec/build.gradle (hunk 1 above)
Comment (ayeshLK) [reply docs/spec/build.gradle:21]: Agreed +1. As an improvement we can use the Ballerina nightly docker image for this build IMO. @NipunaRanasinghe WDYT ?

File: docs/spec/sanitations.bal [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        } else if value.post !is () {
+            responses = value?.post?.responses;
+        }
+
+        if responses !is map<ResponseCode> {
+            continue;
+        }
+        foreach [string, ResponseCode] [_, item] in responses.entries() {
+            map<ResponseHeader> content = item.content ?: {};
+            ResponseHeader app = content[CONTENT_TYPE_JSON] ?: {};
```
Comment (ayeshLK) [inline docs/spec/sanitations.bal:92]: I think we can do this. 
```suggestion
            ResponseHeader app = content[mime:APPLICATION_JSON] ?: {};
```

File: docs/spec/build.gradle (hunk 1 above)
Comment (NipunaRanasinghe) [reply docs/spec/build.gradle:21]: @ayeshLK it seems we currently dont support single files in the Ballerina gradle plugin. Confirmed with @ThisaruGuruge and @MohamedSabthar as well.

Btw I still believe your point is valid and ideally all the Ballerina build and run processes should go through the Ballerina gradle plugin, specially to make sure that we use the same version across all the ballerina builds within a given connector project.

Comment (ayeshLK) [reply docs/spec/build.gradle:21]: +1

File: docs/spec/sanitations.bal (hunk 5 above)
Comment (NipunaRanasinghe) [reply docs/spec/sanitations.bal]: Updated in https://github.com/ballerina-platform/module-ballerinax-slack/pull/164/commits/484cc04708e1d473ba55cb9e630776bb8aa7db9f

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**